src/scss/components/switch/mixins/_hxSwitchControl.scss

```diff
@@ -0,0 +1,154 @@
+//
+// ===== Mixins for <hx-switch-control> states =====
+//
+
+// ========== PRISTINE ==========
+@mixin __switchControl($pseudo-state: null, $checkBox-state: null) {
+  @if $pseudo-state == focus {
+    box-shadow: $focus-glow;
+
+    &::-ms-check {
+      border-color: $cyan-700;
+    }
+  } @else if $pseudo-state == hover {
+    @if $checkBox-state == unchecked {
+      background-color: $gray-0;
+      border: 0.125rem solid $cyan-500;
+
+      &::before {
+        background-color: $gray-0;
+        border: 0.125rem solid $cyan-500;
+        top: 0.01rem;
+        left: 0.01rem;
+      }
+
+      &::after {
+        color: $cyan-500;
+      }
+    } @else {
+      background-color: $cyan-500;
+      border: 0.125rem solid $cyan-500;
+
+      &::before {
+        background-color: $gray-0;
+        top: 0.01rem;
+      }
+
+      &::after {
+        color: $gray-0;
+      }
+    }
+  } @else {
+    color: $gray-500;
+
+    &::-ms-check {
+      background-color: $gray-0;
+      border-color: currentColor;
+      border-style: solid;
+      border-width: 0.063rem;
+      color: inherit;
+    }
+  }
+}
+
+// ========== Error ==========
+@mixin __switchControl--error($pseudo-state: null, $checkBox-state: null) {
+  @if $pseudo-state == focus {
+    box-shadow: $focus-glow-invalid;
+
+    &::-ms-check {
+      border-color: $red-900;
+    }
+  } @else if $pseudo-state == hover {
+    &::-ms-check {
+      background-color: $red-100;
+    }
+  } @else {
+    @if $checkBox-state == checked {
+      background-color: $red-900 !important;
+      border: 0.125rem solid $red-900;
+
+      &:after {
+        color: $gray-0 !important;
+      }
+
+      &:before {
+        background-color: $gray-0;
+        border: 0.125rem solid $red-900;
+      }
+    } @else {
+      background-color: $gray-0;
+      border: 0.125rem solid $red-900 !important;
+
+      &::after {
+        color: $red-900 !important;
+      }
+
+      &::before {
+        background-color: $gray-0;
+        border: 0.125rem solid $red-900 !important;
+        top: 0.031rem;
+        left: 0.01rem;
+        content: "\0021";
+        color: $red-900;
+        font-size: 1.063rem;
+        text-align: center;
+      }
+    }
+  }
+}
+
+// ========== DISABLED ==========
+@mixin __switchControl--disabled($pseudo-state: null) {
+  @if $pseudo-state == focus {
+    &::-ms-check {
+      border-color: $gray-500;
+      box-shadow: $focus-glow-disabled;
+    }
+  } @else if $pseudo-state == hover {
+    &::-ms-check {
+      background-color: $gray-100;
+      border-color: $gray-500;
+      color: $gray-500;
+    }
+  } @else if $pseudo-state == unchecked {
+    background-color: $gray-200;
```

Comment:
Please indent this block.

Reply:
As per our lint rules, `yarn lint` is not throwing any error. So, i continued the same.